### PR TITLE
feat: add setters and getters for da and dam

### PR DESCRIPTION
**Pull request by tadejsv**
This adds setter and getter methods for DA and DAM, for `texts` and `tags`

### CODE REVIEW

File: jina/types/arrays/memmap.py [hunk 1]
```diff
@@ -563,6 +564,22 @@ def embeddings(self, emb: np.ndarray):
...
+        """
+        return self.get_attributes('tags')
+
+    @DocumentArrayGetAttrMixin.texts.getter
+    def texts(self) -> Tuple[str]:
+        """Get the text attribute of all Documents
+
+        :return: List of ``text`` attributes for all Documents
+        """
+        return self.get_attributes('tags')
```
Comment (numb3r3) [inline jina/types/arrays/memmap.py:581]:
```suggestion
        return self.get_attributes('texts')
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: tests/unit/types/arrays/test_memmap.py [hunk 2]
```diff
@@ -517,3 +517,57 @@ def test_blobs_setter_dma():
     np.testing.assert_almost_equal(da.blobs, blobs)
     for x, doc in zip(blobs, da):
         np.testing.assert_almost_equal(x, doc.blob)
+
+
+def test_tags_getter_da(tmpdir):
```
Comment (alaeddine-13) [inline tests/unit/types/arrays/test_memmap.py:522]:
```suggestion
def test_tags_getter_dam(tmpdir):
```

File: tests/unit/types/arrays/test_memmap.py [hunk 3]
```diff
@@ -517,3 +517,57 @@ def test_blobs_setter_dma():
...
+    dam = DocumentArrayMemmap(tmpdir)
+    dam.extend([Document(tags={'a': 2, 'c': 'd'}) for _ in range(100)])
+    assert len(dam.tags) == 100
+    assert dam.tags == dam.get_attributes('tags')
+
+
+def test_tags_setter_da(tmpdir):
```
Comment (alaeddine-13) [inline tests/unit/types/arrays/test_memmap.py:529]:
```suggestion
def test_tags_setter_dam(tmpdir):
```

File: tests/unit/types/arrays/test_memmap.py [hunk 4]
```diff
@@ -517,3 +517,57 @@ def test_blobs_setter_dma():
...
+def test_setter_wrong_len(tmpdir):
+    dam = DocumentArrayMemmap(tmpdir)
+    dam.extend([Document() for _ in range(100)])
+    tags = [{'1': 2}]
+
+    with pytest.raises(ValueError, match='the number of tags in the'):
+        dam.tags = tags
+
+
+def test_texts_getter_da(tmpdir):
```
Comment (alaeddine-13) [inline tests/unit/types/arrays/test_memmap.py:549]:
```suggestion
def test_texts_getter_dam(tmpdir):
```

File: tests/unit/types/arrays/test_memmap.py [hunk 5]
```diff
@@ -517,3 +517,57 @@ def test_blobs_setter_dma():
...
+    dam = DocumentArrayMemmap(tmpdir)
+    dam.extend([Document(text='hello') for _ in range(100)])
+    assert len(dam.texts) == 100
+    assert dam.texts == dam.get_attributes('text')
+
+
+def test_texts_setter_da(tmpdir):
```
Comment (alaeddine-13) [inline tests/unit/types/arrays/test_memmap.py:556]:
```suggestion
def test_texts_setter_dam(tmpdir):
```

**Review by alaeddine-13 — APPROVED** [review]
LGTM, just minor fixes

**Review by hanxiao — APPROVED** [review]

**Pull request merged**